src/main/java/Main.java

```diff
@@ -414,6 +420,32 @@ private static File extractFromJar(String resource, String fileName, String suff
         return tmp;
     }
 
+    private static void deleteWinstoneTempContents(File folder, final String...patterns) throws IOException {
+        File tmpdir = folder;
+
+        if(tmpdir == null){
```

Comment:
This is impossible. We already did a null check.

So delete this block, and you can use `folder` directly, not `tmpdir`.

Also BTW it is very dangerous to have a method overload which consists merely of adding varargs to an existing overload, since it is not obvious which overload you are calling when there are zero additional arguments. Rename, or maybe just inline this into its call site since it would be short anyway.